### PR TITLE
Cell Finetuned Model + Tests + OOP

### PR DESCRIPTION
**Pull request by ian-coccimiglio**
I put in a new finetuned object detection model and recipe called  "ObjectAwareModel_Cell_FT" and "Mobile SAM Finetuned" respectively. It has a FasterRCNN_MobileNet backbone, and was finetuned on the entirety of the Cellpose training images and labels (after converting the ground-truth segmentations to bounding boxes) for a few epochs until I got tired of waiting. Still probably plenty of room to optimize, but it's a start.

The model performs respectably well on "cellular" images, whereas the original MobileSAMv2 (pretrained YOLO) model would often fail. However, it simultaneously performs worse on "general" images.

I also changed part of the code structure to be more object-oriented by creating some classes for object detectors. This was because I was starting to add repetitive functions every time I wanted to work on a new model. However, the API remains nearly the same, with `get_bounding_boxes` as being the major driver of object_detection, but now you need to select the detector_model between "YOLOv8" and "Finetuned". Names are only placeholders.

### CODE REVIEW

**Comment by bnorthan** [comment]
Thanks - this project is really starting to look great.
